src/renderer/modals/VoteTron/fields/VotesField.js

```diff
@@ -213,48 +230,138 @@ const SRRow = ({ sr, rank, isSR, value, disabled, max, onChange, onExternalLink
         </SubTitle>
       </InfoContainer>
       <InputBox active={!!value}>
-        <IconContainer onClick={onReset}>
+        <IconContainer
+          onClick={() => dispatch({ type: "updateVote", address: sr.address, value: "" })}
+        >
           <Cross size={16} />
         </IconContainer>
         <VoteInput
-          /** $FlowFixMe */
+          // $FlowFixMe
           ref={inputRef}
-          defaultValue={value}
-          disabled={!value && disabled}
+          placeholder={"0"}
           min={0}
-          {...(max > 0 ? { max } : {})}
-          onChange={e => onChange(sr.address, e.target.value)}
+          max={null}
+          type="text"
+          value={value}
+          disabled={!value && disabled}
+          onChange={e =>
+            dispatch({ type: "updateVote", address: sr.address, value: e.target.value })
+          }
         />
       </InputBox>
     </Row>
   );
 };
 
+const SRRow = memo(_SRRow);
+
 const SR_THRESHOLD = 27;
 const SR_MAX_VOTES = 5;
 
+const fuseOptions = {
+  threshold: 0.1,
+  keys: ["name", "address"],
+  shouldSort: true,
+};
+
+const useSortedSr = (
+  search: string,
+  superRepresentatives: SuperRepresentative[],
+  votes: Vote[],
+) => {
+  const [initialVotes] = useState(votes.map(({ address }) => address));
+
+  const fuse = useMemo(
+    () =>
+      new Fuse(
+        superRepresentatives.map((sr, rank) => ({
+          sr,
+          name: sr.name,
+          address: sr.address,
+          rank: rank + 1,
+          isSR: rank <= SR_THRESHOLD,
+        })),
+        fuseOptions,
+      ),
+    [superRepresentatives],
+  );
+
+  const initSort = useCallback(
+    ({ sr: { address }, rank }, { sr: { address: _address }, rank: _rank }) =>
+      initialVotes.includes(address)
+        ? initialVotes.includes(_address)
+          ? rank - _rank
+          : -1
+        : initialVotes.includes(_address)
+        ? rank - _rank
+        : 0,
+    [initialVotes],
+  );
+
+  const sr = useMemo(
+    () =>
+      search
+        ? fuse.search(search)
+        : superRepresentatives
+            .map((sr, rank) => ({ sr, rank: rank + 1, isSR: rank <= SR_THRESHOLD }))
+            .sort(initSort),
```

Comment:
I feel we don't need to involve any sort here but it's just

```
initialVotes.map(addr => lookup by address)
.concat(superRepresentatives minus the looked up)
```

this could simplify a lot of the algorithm here..